### PR TITLE
[PyROOT] Avoid linking TPython against libPython

### PR DESCRIPTION
**Pull request by vepadulano**
I'm opening this PR to give it a go with our CI. This change is somewhat necessary on many levels (conda benefits from it, pip requires it). And it is also a good habit in general not to link against libPython.

EDIT: possibly superseeded by https://github.com/root-project/root/pull/16337

### CODE REVIEW

**Comment by dpiparo** [comment]
Good move. The CI is quite demanding these days. If the change works there AND in the Conda CI, the coverage should be rather good. Clearly, we should add a note in the RN to inform owners of large stacks, e.g. LHC experiments, of this change in case something behaves differently in that context.

**Comment by vepadulano** [comment]
> e.g. LHC experiments, of this change in case something behaves differently in that context.

This move is necessary for the Python environments but there is a clear limitation in its current form, that is it breaks the usage of TPython from C++:

```
root.exe -l -b -q -x -e 'TPython::Exec("print(\"1 + 1 =\", 1+1)")'

cling::DynamicLibraryManager::loadLibrary(): dlopen(/Users/vpadulan/Programs/rootproject/rootbuild/master-4e3ca10195-pyroot-debug/lib/libROOTTPython.so, 0x0009): symbol not found in flat namespace '_PyBool_Type'
Error in <AutoloadLibraryMU>: Failed to load library /Users/vpadulan/Programs/rootproject/rootbuild/master-4e3ca10195-pyroot-debug/lib/libROOTTPython.socling JIT session error: Failed to materialize symbols: { (main, { __ZN7TPython4ExecEPKc }) }
```

This is not surprising, we are purposely removing the linking against libPython so TPython cannot find the symbols. From within a Python interpreter, libpython is automatically injected and linked at the global scope, but that doesn't happen when a symbol from libpython is needed from an executable outside of Python itself.

Bottom line, we will need to investigate how to properly manage both the requirements of Python packaging systems and embedding libpython in other cases.

**Comment by guitargeek** [comment]
I think the sustainable fix is to get rid of TPython all together, since the provided functionality of the `CPyCppyy` API *should* be the same.

In fact, there was a time where TPython was completely gone from the codebase until Enric brought it back with 2934be5fb3007.

Maybe it is now a good time to complete the migration to the `CPyCppyy` API?

**Comment by guitargeek** [comment]
FYI, I have opened some PRs for the CPyCppyy API to make it fit for replacing TPython:
  * https://github.com/wlav/CPyCppyy/pull/30
  * https://github.com/wlav/CPyCppyy/pull/31

**Comment by vepadulano** [comment]
Thank you @guitargeek I believe this PR can be closed now!
